pkg/statistics/handle/ddl/ddl.go

```diff
@@ -116,13 +118,36 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 			}
 		}
 	case model.ActionDropTablePartition:
-		// TODO: Update the modify count and count for the global table.
-		_, droppedPartitionInfo := t.GetDropPartitionInfo()
+		globalTableInfo, droppedPartitionInfo := t.GetDropPartitionInfo()
+
+		delta := int64(0)
+		count := int64(0)
 		for _, def := range droppedPartitionInfo.Definitions {
+			// Get the count and modify count of the partition.
+			stats := h.statsHandler.GetPartitionStats(globalTableInfo, def.ID)
+			if stats.Pseudo {
+				logutil.StatsLogger.Warn(
+					"drop partition with pseudo stats, "+
+						"usually it won't happen because we always load stats when initializing the handle",
+					zap.String("table", globalTableInfo.Name.O),
+					zap.String("partition", def.Name.O),
```

Comment:
I think db name should also be printed, but looks like it would take much extra work to implement that.

Reply:
We need to pass the info schema here. And it involves a lot of test changes. I will do it in a separate PR.

Reply:
See https://github.com/pingcap/tidb/pull/49011